test/IRGen/big_types_corner_cases.swift

```diff
@@ -271,3 +271,60 @@ class ClassWithLoadableStructWithBiggerString {
         return LoadableStructWithBiggerString(a1: BiggerString(str:"", double:0.0), a2: [], a3: [])
     }
 }
+
+//===----------------------------------------------------------------------===//
+// SR-8076
+//===----------------------------------------------------------------------===//
+public struct sr8076_BigStruct {
+    public var m1: String
+    public var m2: String
+    public var m3: String
+}
+
+public typealias sr8076_Filter = (sr8076_BigStruct) -> Bool
+
+public protocol sr8076_Query {
+    associatedtype Returned
+}
+
+public protocol sr8076_ProtoQueryHandler {
+    func forceHandle_1<Q: sr8076_Query>(query: Q) -> Void
+    func forceHandle_2<Q: sr8076_Query>(query: Q) -> (Q.Returned, sr8076_BigStruct?)
+    func forceHandle_3<Q: sr8076_Query>(query: Q) -> (Q.Returned, sr8076_Filter?)
+    func forceHandle_4<Q: sr8076_Query>(query: Q) throws -> (Q.Returned, sr8076_Filter?)
+}
+
+public protocol sr8076_QueryHandler: sr8076_ProtoQueryHandler {
+    associatedtype Handled: sr8076_Query
+    func handle_1(query: Handled) -> Void
+    func handle_2(query: Handled) -> (Handled.Returned, sr8076_BigStruct?)
+    func handle_3(query: Handled) -> (Handled.Returned, sr8076_Filter?)
+    func handle_4(query: Handled) throws -> (Handled.Returned, sr8076_Filter?)
+}
+
+public extension sr8076_QueryHandler {
```

Comment:
Good tests. but, can you please add `CHECK`(s) that make sure the resulting IR is correct? right now we are only making sure the compiler doesn't crash, but, we are not looking at the IR to see that it makes sense. Look at some of the `CHECK` and `CHECK-LABEL` lines inside this very same test file for examples.

Reply:
Fixed.